MariaDB/server
# MDEV-33852: Rework systemd installation on Debian

- [x] *The Jira issue number for this PR is: MDEV-33852*


## Description
Let dh_systemd handle most of the systemd side and get rid of custom scripts

Rework installation of systemd service and socket files base on Michael Biebl merge request:

https://salsa.debian.org/mariadb-team/mariadb-server/-/merge_requests/63
https://salsa.debian.org/mariadb-team/mariadb-server/-/merge_requests/75

## Release Notes
They should no be mentioned in Release Notes they should not affect anything


## How can this PR be tested?
Compile DEB packages. Install them. check if `mariadbd` is running. Remove packages and check that systemd files have been removed and `mariadbd` is not running anymore


## Basing the PR against the correct MariaDB version

## PR quality check
